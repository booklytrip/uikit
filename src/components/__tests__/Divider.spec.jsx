import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Divider from '../Divider';

describe('Divider', () => {
    it('renders without crashing', () => {
        mount(<Divider />);
    });

    describe('Snapshots', () => {
        it('renders with vertical divider', () => {
            const rendered = renderer.create(<Divider vertical />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with horizontal divider', () => {
            const rendered = renderer.create(<Divider horizontal />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders hidden divider', () => {
            const rendered = renderer.create(<Divider hidden />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders fitted divider', () => {
            const rendered = renderer.create(<Divider fitted />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
