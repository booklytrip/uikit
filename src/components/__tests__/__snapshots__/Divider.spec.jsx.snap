exports[`Divider Snapshots renders fitted divider 1`] = `
<div
  className="ui divider fitted" />
`;

exports[`Divider Snapshots renders hidden divider 1`] = `
<div
  className="ui divider hidden" />
`;

exports[`Divider Snapshots renders with horizontal divider 1`] = `
<div
  className="ui divider horizontal" />
`;

exports[`Divider Snapshots renders with vertical divider 1`] = `
<div
  className="ui divider vertical" />
`;
